File: app/src/main/java/com/example/snaptaplaque/models/api/account/RegisterRequest.java
package com.example.snaptaplaque.models.api.account;

import com.google.gson.annotations.SerializedName;

/**
 * Représente une requête d'inscription d'utilisateur.
 */
public class RegisterRequest {

    /**
     * Adresse email de l'utilisateur.
     */
    @SerializedName("email")
    private String email;

    /**
     * Nom d'utilisateur choisi par l'utilisateur.
     */
    @SerializedName("username")
    private String username;

    /**
     * Mot de passe choisi par l'utilisateur.
     */
    @SerializedName("password")
    private String password;

    /**
     * Nom complet de l'utilisateur (prénom + nom).
     */
    @SerializedName("full_name")
    private String full_name;

    /**
     * Indique si l'utilisateur est un administrateur (true) ou un utilisateur standard (false).
     */
    @SerializedName("is_admin")
    private boolean is_admin;

    /**
     * Indique si l'utilisateur à donner son consentement au RGPD (true) ou non (false).
     */
    @SerializedName("gdpr_consent")
    private boolean gdpr_consent;

    /**
     * Constructeur de la classe RegisterRequest.
     *
     * @param email L'adresse email de l'utilisateur.
     * @param username Le nom d'utilisateur choisi par l'utilisateur.
     * @param password Le mot de passe choisi par l'utilisateur.
     * @param full_name Le nom complet de l'utilisateur (prénom + nom).
     * @param is_admin Indique si l'utilisateur est un administrateur (true) ou un utilisateur standard (false).
     * @param gdpr_consent Indique si l'utilisateur à donner son consentement au RGPD (true) ou non (false).
     */
    public RegisterRequest(String email,
                           String username,
                           String password,
                           String full_name,
                           boolean is_admin,
                           boolean gdpr_consent) {
        this.email = email;
        this.username = username;
        this.password = password;
        this.full_name = full_name;
        this.is_admin = is_admin;
        this.gdpr_consent = gdpr_consent;
    }

    public String getEmail() {
        return email;
    }

    public String getUsername() {
        return username;
    }

    public String getPassword() {
        return password;
    }

    public String getFull_name() {
        return full_name;
    }

    public boolean is_admin() {
        return is_admin;
    }

    public boolean is_gdpr_consent() {
        return gdpr_consent;
    }

}

File: iOS/SnapTaPlaque/Models/API/Account/RegisterRequest.swift
//
//  RegisterRequest.swift
//  SnapTaPlaque
//
//  Created by Victor Poirier on 20/03/2026.
//

import Foundation

/// Représente la demande d'inscription d'un nouvel utilisateur au service "SnapTaPlaque".
///
/// Ce modèle encode les informations requises pour créer un compte via l'API, en transformant
/// les propriétés locales en JSON (format `snake_case`) grâce à l'implémentation de `Codable`.
struct RegisterRequest: Codable {
    
    /// L'adresse courriel de l'utilisateur.
    let email: String
    
    /// Le nom d'utilisateur unique qui servira d'identifiant de connexion.
    let username: String
    
    /// Le mot de passe en texte clair, choisi par l'utilisateur.
    ///
    /// - Important: Doit être envoyé à l'API impérativement via une connexion chiffrée (HTTPS).
    let password: String
    
    /// Le nom complet (nom et prénom) de l'utilisateur.
    let fullName: String
    
    /// Détermine si l'utilisateur fraîchement créé disposera des privilèges d'administration.
    /// Par défaut, cette valeur devrait toujours être transmise à `false` côté client pour des raisons de sécurité.
    let isAdmin: Bool
    
    /// Témoigne du consentement explicite de l'utilisateur concernant le traitement de ses données
    /// personnelles (obligatoire au regard du RGPD).
    let gdprConsent: Bool
    
    /// Gère la nomenclature des clés lors de l'encodage JSON à destination de l'API.
    enum CodingKeys: String, CodingKey {
        case email
        case username
        case password
        
        /// Encodé en tant que `full_name`.
        case fullName = "full_name"
        
        /// Encodé en tant que `is_admin`.
        case isAdmin = "is_admin"
        
        /// Encodé en tant que `gdpr_consent`.
        case gdprConsent = "gdpr_consent"
    }
}
